src/components/VideoCard/VideoCard.jsx

```diff
@@ -19,8 +22,17 @@ const VideoCard = ({ video }) => {
         );
     };
 
+    const navigate = useNavigate();
+
     return (
-        <div className="video-card-container" key={video._id}>
+        <div
+            className="video-card-container"
+            key={video._id}
+            onClick={() => {
+                navigate(`/watch/${video._id}`);
+                setHistoryHandler(video);
+            }}
+        >
```

Comment:
Since you are using the `navigate()` on the `onClick`, your `OverflowMenu` can't be accessed and clicking on it just navigates to the video listing page.

Reply:
It is fixed now. Thanks for pointing it out!